File: modules/scala/futureOpenTracing/src/main/java/io/opentelemetry/javaagent/instrumentation/scalafuture/ScalaFutureInstrumentationModule.java
package io.opentelemetry.javaagent.instrumentation.scalafuture;

import static java.util.Arrays.asList;

import com.google.auto.service.AutoService;
import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
import java.util.List;

@AutoService(InstrumentationModule.class)
public class ScalaFutureInstrumentationModule extends InstrumentationModule {

    public ScalaFutureInstrumentationModule() {
        super("scala-future", "scala-future-2.10", "scala-concurrent");

        System.out.println("carsarecool");
    }

    @Override
    public List<TypeInstrumentation> typeInstrumentations() {
        return asList(
                new ScalaCallbackRunnableInstrumentation(),
                new ScalaPromiseTransformationInstrumentation());
    }
}

File: modules/scala/futureOpenTracing/src/main/scala/io/opentelemetry/javaagent/instrumentation/scalafuture/ScalaFutureInstrumentationModule.scala
package io.opentelemetry.javaagent.instrumentation.scalafuture

import io.opentelemetry.javaagent.extension.instrumentation.{InstrumentationModule, TypeInstrumentation}

import java.util

class ScalaFutureInstrumentationModule
    extends InstrumentationModule("scala-future", "scala-future-2.10", "scala-concurrent") {

  override def typeInstrumentations(): util.List[TypeInstrumentation] = {
    // uses java as scala convertors are not available and cause wierd no-arg constructor errors.
    util.Arrays.asList(
      new ScalaCallbackRunnableInstrumentation,
      new ScalaPromiseTransformationInstrumentation
    )
  }
}
